### ISSUE TITLE
Bulk Delete for All Activity in Specific Month

### ISSUE THREAD

**Issue by rafinkanisa** [iMMAP/ngm-reportDesk#74]
**Is your feature request related to a problem? Please describe.**
Some users are requesting to delete all of the activities they have reported in a certain month at ReportHub. Currently, ReportHub doesn't have the functionality to delete all activities either that have been saved as drafts or have been submitted using a single button. I would like to decrease the developer task load to delete such reports. It would be better for the reporting partners to be able to do it themselves rather than asking through developer team to delete it directly from the DB. Deleting directly from the DB also imposed on high risk.

**Describe the solution you'd like**
Create a single button to delete all activities in a certain month

**Acceptance Criteria**

- [x] Show delete button in the report month for the first time
- [x] Show the delete button in the report month in edit mode
- [x] Delete button should appear at the top right corner
- [x] Delete button appear at the bottom along with the save
- [x] If it's grouped by Province the delete button will show up after clicking a certain province
- [x] If it's grouped by District the delete button will show up after clicking a certain district
- [x] Clicking the delete button will delete all of the activities on the selected month
- [x] A prompt message will show up to confirm whether the user would like to delete all of the activities
- [x] The delete button should only appear to the focal point and admin of the organization (our account should also have it Naqi, me, and you)

**Describe alternatives you've considered**

**Additional context**
Please use below terminology

Button:
Delete All Activities

Prompt Messages:
Header: Delete All Activities
Body: Are you sure you want to DELETE ALL ACTIVITIES in XX report?
Warning: Once you delete the activities, Any of the lost data will not be accessible anymore and this action can't be recovered by the ReportHub Team.

XX = month of the report (e.g. February and so on)

**Comment by fakhrihawari** [comment]
Mock Up
First Put in Top
<img width="1255" alt="Screen Shot 2022-04-13 at 14 04 03" src="https://user-images.githubusercontent.com/12655589/163119319-9a623716-e551-41fd-b550-fcd14e1f797f.png">

on the bottom
<img width="1264" alt="Screen Shot 2022-04-13 at 14 04 09" src="https://user-images.githubusercontent.com/12655589/163119425-e36213b4-5565-4e23-9bce-e30cb3375c84.png">


Modal
<img width="933" alt="Screen Shot 2022-04-13 at 14 06 34" src="https://user-images.githubusercontent.com/12655589/163119543-992a24df-4583-4caf-b442-fc01f996bc81.png">

**Comment by rafinkanisa** [comment]
Please edit the words used as described.

A quick question: when a user is deleting the activities, this action must have to be saved first before it will be taking an action at the platform right?

**Comment by fakhrihawari** [comment]
No need because I set the function to automatically save the report after we delete all the activities

**Comment by fakhrihawari** [comment]
Button
<img width="1335" alt="Screen Shot 2022-04-14 at 10 55 46" src="https://user-images.githubusercontent.com/12655589/163310777-7c95aaff-bbb4-4e78-b359-eb43eed5974c.png">
<img width="1325" alt="Screen Shot 2022-04-14 at 10 55 40" src="https://user-images.githubusercontent.com/12655589/163310785-325d661d-e76b-4f10-9d9d-2ede38fa228e.png">

Prompt message
<img width="1021" alt="Screen Shot 2022-04-14 at 10 57 20" src="https://user-images.githubusercontent.com/12655589/163310957-cc36f90b-e273-404f-b3ae-0b81180cf975.png">

**Comment by fakhrihawari** [comment]
updated on DEV

**Comment by rafinkanisa** [comment]
Hi I've just checked and found this. Why are there 2 button (delete activities & remove report)

<img width="578" alt="Screen Shot 2022-04-20 at 14 38 07" src="https://user-images.githubusercontent.com/28089698/164176027-51ae6eb7-6231-4da5-9a46-a7149f8acff5.png">

I'm checking using common user
this is the project link
https://dev.reporthub.org/desk/#/cluster/projects/report/612f231d7fd758f3603104ef/612f231d7fd758f3603104f6

**Comment by fakhrihawari** [comment]
The "remove report" button is used to delete non-blank reports that are outside the project date range that has been defined in the project plan.
So for example, when we first create a project, we set the project to start on January 1, 2021, and end on December 31, 2022.

Later on January 1, 2022, we changed the project start date to January 1, 2022, for some reason.
Reports that have been created before 2022 will be deleted if the report is blank automatically. On the other hand, non-blank reports are not deleted. 
 It is because non-blank reports maybe contain important information  
So if we want to delete the non-blank reports we need to click these "remove report" buttons.
